runtime/drivers/duckdb/olap.go

```diff
@@ -549,6 +609,61 @@ func (c *connection) RenameTable(ctx context.Context, oldName, newName string, v
 	return err
 }
 
+func (c *connection) execIncrementalInsert(ctx context.Context, safeName, sql string, byName bool, strategy drivers.IncrementalStrategy, uniqueKey []string) error {
+	var byNameClause string
+	if byName {
+		byNameClause = "BY NAME"
+	}
+
+	if strategy == drivers.IncrementalStrategyAppend {
+		return c.execWithLimits(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("INSERT INTO %s %s (%s\n)", safeName, byNameClause, sql),
+			Priority:    1,
+			LongRunning: true,
+		})
+	}
+
+	if strategy == drivers.IncrementalStrategyMerge {
+		// Create a temporary table with the new data
+		tmp := uuid.New().String()
+		err := c.execWithLimits(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("CREATE TEMPORARY TABLE %s AS (%s\n)", safeSQLName(tmp), sql),
+			Priority:    1,
+			LongRunning: true,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Drop the rows from the target table where the unique key is present in the temporary table
+		where := ""
+		for i, key := range uniqueKey {
+			key = safeSQLName(key)
+			if i != 0 {
+				where += " AND "
+			}
+			where += fmt.Sprintf("base.%s = tmp.%s", key, key)
+		}
+		err = c.execWithLimits(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("DELETE FROM %s base WHERE EXISTS (SELECT 1 FROM %s tmp WHERE %s)", safeName, safeSQLName(tmp), where),
+			Priority:    1,
+			LongRunning: true,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Insert the new data into the target table
+		return c.execWithLimits(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("INSERT INTO %s %s SELECT * FROM %s", safeName, byNameClause, safeSQLName(tmp)),
+			Priority:    1,
+			LongRunning: true,
+		})
+	}
+
+	return fmt.Errorf("incremental insert strategy %q not supported", strategy)
+}
```

Comment:
**Issue**: The method 'execIncrementalInsert' introduces several potential issues related to SQL injection and error handling. The dynamic construction of SQL queries using parameters that could be manipulated poses a risk of SQL injection. Moreover, the method lacks comprehensive error handling and logging, which are crucial for debugging and maintaining the code. <br> **Fix**: To mitigate SQL injection risks, use parameterized queries or prepared statements instead of string formatting for SQL commands. Implement thorough error handling and logging to ensure that all potential issues are captured and can be addressed promptly. Refactor the method to improve readability and maintainability. <br> **Code Suggestion**: 
 ```
 -func (c *connection) execIncrementalInsert(ctx context.Context, safeName, sql string, byName bool, strategy drivers.IncrementalStrategy, uniqueKey []string) error {
 +func (c *connection) execIncrementalInsert(ctx context.Context, safeName string, sql string, byName bool, strategy drivers.IncrementalStrategy, uniqueKey []string) error {
 +	stmt, err := c.prepare(fmt.Sprintf("INSERT INTO %s (%s) VALUES (?)", safeName, strings.Join(uniqueKey, ", ")))
 +	if err != nil {
 +		return err
 +	}
 +	defer stmt.Close()
 +	_, err = stmt.Exec(sql)
 +	if err != nil {
 +		return err
 +	}
 +	return nil
 ```